### DISCUSSION TITLE
Is there any way to build Raspbian artifacts in actions?

### DISCUSSION THREAD

**Discussion by brunchboy** [community/community#25650] [Actions]
<p>I recently started using GitHub Actions to automate the creation of release artifacts on each code push, and have been delighted to produce all three of the deliverables for my most popular project, which are an executable Jar file, a windows MSI installer, and a code signed and notarized Mac disk image. This is an amazing platform, thanks so much!</p>
<p>Now I am wondering if I can achieve something similar for another popular project, but one of the formats I need to build is a Rasbpian binary. Is there any way, even with docker jobs, to run and build code for the ARMv6 architecture? Or is that not yet practical until it becomes a supported runner type?</p>

**Comment by Yanjingzhu** [comment]
<p>GitHub Actions now supports self-hosted runners on the ARM and ARM64 architectures. <a href="https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.blog%2Fchangelog%2F2019-12-03-github-actions-self-hosted-runners-on-arm-architectures%2F&amp;data=02%7C01%7Cv-yjingz%40microsoft.com%7C6a17e73958c24a3caca508d797f7d6be%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637144963049697725&amp;sdata=TSoK8Anp9TXtK4%2FHv2wMxAVN%2B%2FC2pnVLmF71KgW0%2FCY%3D&amp;reserved=0" rel="nofollow noopener">https://github.blog/changelog/2019-12-03-github-actions-self-hosted-runners-on-arm-architectures/</a></p>
<p>You could consider to set up a self-hosted runner on your ARMv6 architecture machine.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/a/a027f20fe430872a26e9f1a1e46dfb195fbf4450.png" data-download-href="/uploads/short-url/mQNYEjhSQ9qFvbqrNxPyIFAkCpG.png?dl=1" title="self-hosted runner.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181092352-54b0f7ad-f907-4ff5-8607-1c3bc5e3b76a.png" alt="self-hosted runner.png" data-base62-sha1="mQNYEjhSQ9qFvbqrNxPyIFAkCpG" width="687" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/a/a027f20fe430872a26e9f1a1e46dfb195fbf4450_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">self-hosted runner.png</span><span class="informations">711×517 27.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by brunchboy** [comment]
<p>That’s interesting, but I only have one Raspberry Pi, it is not always connected to the internet, and there are all kinds of security warnings about using self-hosted runners on hardware you care about, so that isn’t the answer I was hoping for. Are there plans to offer GitHub hosted virtual machines with ARM architecture someday?</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p><em>&gt;&gt;Are there plans to offer GitHub hosted virtual machines with ARM architecture someday?</em></p>
<p>Please share this feature requests for GitHub Actions in the <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">Feedback form for GitHub Actions</a>.</p>

**Comment by Ryazbeck** [comment]
<p>Hello, I tried to install Github Actions Runner on a Pi Zero W with Armv6 and it is failing.</p>
<aside class="onebox githubissue">
  <header class="source">
      <a href="https://github.com/actions/runner/issues/688" target="_blank" rel="nofollow noopener">github.com/actions/runner</a>
  </header>
  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Issue">
	  <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/actions/runner/issues/688" target="_blank" rel="nofollow noopener">Pi Zero W: ./config.sh: line 79:  6312 Segmentation fault ./bin/Runner.Listener configure "$@"</a>
    </h4>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2020-08-29" data-time="22:54:16" data-timezone="UTC">10:54PM - 29 Aug 20 UTC</span>
      </div>


      <div class="user">
        <a href="https://github.com/Ryazbeck" target="_blank" rel="nofollow noopener">
          <img alt="Ryazbeck" src="https://user-images.githubusercontent.com/5881176/181092355-5491b682-e3a0-4c02-b8f9-b8f8a0d80623.jpeg" class="onebox-avatar-inline" width="20" height="20">
          Ryazbeck
        </a>
      </div>
    </div>
  </div>
</div>

<div class="github-row">
  <p class="github-content">Describe the bug
Going through the steps to add self-hosted runner on a Raspberry Pi Zero W. Apologies if this is the...</p>
</div>

<div class="labels">
    <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">bug</span>
</div>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by SiggiSigmann** [comment]
<p>Hi, i have the same problem. I try to use a Pi 2B (I know it is outdated). Did you found a solution. I think a problem could be the limited memory.<br>
Thanks.<br>
Siggi</p>

**Comment by Vincent-Stragier** [comment]
<p>Hi,</p>
<p>I don’t know if you are looking still looking for some solution, but I was able after some (a lot) of trial and error to build a project using the arm-runner-action from the marketplace. I had compiling cmake (which takes time) since I needed it, and it seems to be the only way to make it run in qemu for the moment (there is a bug in cmake and the last version is not yet in Raspberry Pi packages repository). You can have a look at the action I used here: <a href="https://github.com/eLAB-Birdhouse/birdhouse_raspimjpeg/actions/runs/1155636683/workflow" class="inline-onebox" rel="noopener nofollow ugc">Add image to release · eLAB-Birdhouse/birdhouse_raspimjpeg@514b07e · GitHub</a></p>
<p>Note that I did not fully test my binary after building it… So maybe it’s not fully working.</p>
<p>Vincent</p>
